src/library/scala/collection/immutable/Stream.scala

```diff
@@ -25,7 +25,11 @@ import Stream.cons
 
 @deprecated("Use LazyList (which is fully lazy) instead of Stream (which has a lazy tail only)", "2.13.0")
 @SerialVersionUID(3L)
-sealed abstract class Stream[+A] extends AbstractSeq[A] with LinearSeq[A] with LinearSeqOps[A, Stream, Stream[A]] with Serializable {
+sealed abstract class Stream[+A] extends AbstractSeq[A]
+  with LinearSeq[A]
+  with LinearSeqOps[A, Stream, Stream[A]]
+  with IterableFactoryDefaults[A @uncheckedVariance, Stream]
```

Comment:
The problem with doing it once in a central spot is that it has far-reaching consequences (as this PR shows by making it more explicit) and you're turning off refchecks on the entire thing. I'm also not a fan of turning off variance checking on element types, but by reminding ourselves every time we mix in `IterableFactoryDefaults`, we will hopefully be more vigilant.

Reply:
To summarize what we discussed on the meeting: In the new design of this PR, the compiler actually tells you if you need to override `iterableFactory`, `fromSpecific` and `newBuilder` with a more precise type (because you refined `C` / `CC`), where as the protected type alias with uncheckedVariance fools refchecks and no error is issued.

This example gives a class cast exception: https://gist.github.com/lrytz/fe09791208969f07e211c5c8601db125